internal/generator/common/utils_test.go

```diff
@@ -697,3 +697,119 @@ func TestWalkWithFilter(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) { testFunc(t, tc) })
 	}
 }
+
+func TestExtractValuesFromTemplate(t *testing.T) {
+	type testCase struct {
+		name     string
+		template string
+		expected []string
+	}
+
+	testCases := []testCase{
+		{
+			name:     "Empty template",
+			template: "",
+			expected: []string{},
+		},
+		{
+			name:     "Valid template",
+			template: "{{ .foo }}.{{.boo}}",
+			expected: []string{"foo", "boo"},
+		},
+		{
+			name:     "Template with functions",
+			template: "{{ upper .foo | lower }}@{{ .boo }}",
+			expected: []string{"foo", "boo"},
+		},
+		{
+			name:     "Invalid template",
+			template: "{_{ foo }}",
+			expected: []string{},
+		},
+	}
```

Comment:
Based on my comment, you need to add cases to the function under test, at least test cases with the golang template's `index` function.